config: add ability to run code before box.cfg (#11223)
tarantool/tarantool: changelogs/unreleased/gh-10182-add-ability-to-run-code-before-box-cfg.md

Reviewer: Regarding the documentation request: it is important nuance that `config:get()` doesn't work on the preload phase (neither for the app script, nor for modules). I think we should highlight this fact.

Author: I've added this to doc-request.

## feature/config

> * Introduced `preload` tag for roles and scripts, which, when set, will load
> * the role or execute the script before `box.cfg` (gh-10182).

Reviewer 2: ```suggestion
* Introduced a `preload` tag for roles and scripts. When set, it makes Tarantool
* load the role or execute the script before `box.cfg` (gh-10182).
```

Author: Thanks, fixed!

> * load the role or execute the script before `box.cfg` (gh-10182).

Reviewer: Nit: the second asterisk is redundant.

Author: Thanks for noticing! fixed.
